### ISSUE TITLE
onnx 转换失败

### ISSUE THREAD

**Issue by ShuangLiu1992** [Tencent/ncnn#567]
pytorch 0.4.1 中带有reshape的网络转换后无法执行。onnx2ncnn中默认reshape会带有shape，但是现在reshape的实际shape是在另一个Constant中描述，由于缺乏注释我不太清楚reshape的转换具体是怎么执行的。

例子：
graph(%input : Float(64, 1, 28, 28)
      %1 : Float(10, 1, 5, 5)
      %2 : Float(10)
      %3 : Float(20, 10, 5, 5)
      %4 : Float(20)
      %5 : Float(50, 320)
      %6 : Float(50)
      %7 : Float(10, 50)
      %8 : Float(10)) {
  %9 : Float(64, 10, 24, 24) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[5, 5], pads=[0, 0, 0, 0], strides=[1, 1]](%input, %1, %2), scope: Net/Conv2d[conv1]
  %10 : Float(64, 10, 12, 12) = onnx::MaxPool[kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%9), scope: Net
  %11 : Float(64, 10, 12, 12) = onnx::Relu(%10), scope: Net
  %12 : Float(64, 20, 8, 8) = onnx::Conv[dilations=[1, 1], group=1, kernel_shape=[5, 5], pads=[0, 0, 0, 0], strides=[1, 1]](%11, %3, %4), scope: Net/Conv2d[conv2]
  %13 : Float(64, 20, 4, 4) = onnx::MaxPool[kernel_shape=[2, 2], pads=[0, 0, 0, 0], strides=[2, 2]](%12), scope: Net
  %14 : Float(64, 20, 4, 4) = onnx::Relu(%13), scope: Net
  %15 : Dynamic = onnx::Constant[value=  -1  320 [ CPULongTensor{2} ]](), scope: Net
  %16 : Float(64, 320) = onnx::Reshape(%14, %15), scope: Net
  %17 : Float(64, 50) = onnx::Gemm[alpha=1, beta=1, broadcast=1, transB=1](%16, %5, %6), scope: Net/Linear[fc1]
  %18 : Float(64, 50) = onnx::Relu(%17), scope: Net
  %19 : Float(64, 50), %20 : Dynamic = onnx::Dropout[is_test=1, ratio=0.5](%18), scope: Net
  %21 : Float(64, 10) = onnx::Gemm[alpha=1, beta=1, broadcast=1, transB=1](%19, %7, %8), scope: Net/Linear[fc2]
  %output : Float(64, 10) = onnx::Softmax[axis=1](%21), scope: Net
  return (%output);

**Comment by nihui** [comment]
onnx2ncnn 工具已更新，兼容了新 reshape

**Comment by zhyj3038** [comment]
这个问题依旧存在吧!  一个mat经过reshape以后,由于丢失了shape信息,导致得到的mat为空

**Comment by FlyingFishBird** [comment]
mark

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
